src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java

```diff
@@ -586,4 +589,73 @@ FlushService getFlushService() {
   ParameterProvider getParameterProvider() {
     return parameterProvider;
   }
+
+  /** Registers the performance metrics along with JVM and Threads. */
+  private void registerMetricsForClient() {
+    metrics = new MetricRegistry();
+    // Blob histograms
+    blobSizeHistogram = metrics.histogram(MetricRegistry.name("blob", "size", "histogram"));
+    blobRowCountHistogram =
+        metrics.histogram(MetricRegistry.name("blob", "row", "count", "histogram"));
+    cpuHistogram = metrics.histogram(MetricRegistry.name("cpu", "usage", "histogram"));
+
+    // latency metrics
+    flushLatency = metrics.timer(MetricRegistry.name("latency", "flush"));
+    buildLatency = metrics.timer(MetricRegistry.name("latency", "build"));
+    uploadLatency = metrics.timer(MetricRegistry.name("latency", "upload"));
+    registerLatency = metrics.timer(MetricRegistry.name("latency", "register"));
+
+    // throughput metrics
+    uploadThroughput = metrics.meter(MetricRegistry.name("throughput", "upload"));
+    inputThroughput = metrics.meter(MetricRegistry.name("throughput", "input"));
+    SharedMetricRegistries.add(SNOWPIPE_STREAMING_SHARED_METRICS_REGISTRY, metrics);
+
+    JmxReporter jmxReporter =
+        JmxReporter.forRegistry(this.metrics)
+            .inDomain(SNOWPIPE_STREAMING_JMX_METRIC_PREFIX)
+            .convertDurationsTo(TimeUnit.SECONDS)
+            .createsObjectNamesWith(
+                (ignoreMeterType, jmxDomain, metricName) ->
+                    getObjectName(this.getName(), jmxDomain, metricName))
+            .build();
+    jmxReporter.start();
+
+    // add JVM and thread metrics too
+    jvmAndThreadMetrics.register(MetricRegistry.name("jvm", "memory"), new MemoryUsageGaugeSet());
+    jvmAndThreadMetrics.register(MetricRegistry.name("jvm", "threads"), new ThreadStatesGaugeSet());
+
+    SharedMetricRegistries.add(
+        SNOWPIPE_STREAMING_JVM_AND_THREAD_METRICS_REGISTRY, jvmAndThreadMetrics);
```

Comment:
Could we still output these to logger as well, it will be helpful when we can't use jmx (like in temptest, we're looking at logs uploaded to `workflow_logs_v`

Reply:
yes we do it in close() just like before